ext/io/lib.rs

```diff
@@ -81,54 +81,52 @@ deno_core::extension!(deno_io,
   ops = [op_read_sync, op_write_sync],
   esm = [ "12_io.js" ],
   config = {
-    stdio: Rc<RefCell<Option<Stdio>>>,
+    stdio: Option<Stdio>,
   },
   middleware = |op| match op.name {
     "op_print" => op_print::decl(),
     _ => op,
   },
-  state = |state, stdio| {
-    let stdio = stdio
-      .borrow_mut()
-      .take()
-      .expect("Extension only supports being used once.");
-    let t = &mut state.resource_table;
-
-    let rid = t.add(StdFileResource::stdio(
-      match stdio.stdin {
-        StdioPipe::Inherit => StdFileResourceInner {
-          kind: StdFileResourceKind::Stdin,
-          file: STDIN_HANDLE.try_clone().unwrap(),
+  state = |state, cfg| {
+    if let Some(stdio) = cfg.stdio {
```

Comment:
The initialization for this extension was a little strange, but I believe that we can safely avoid the Rc here as state functions are now guaranteed to be `FnOnce`

Reply:
Sounds good, though I think it should be `let stdio = cfg.stdio.unwrap()` - we expect this to be filled in as per the previous version. I'm unsure why it was an option though